graviton/blackbox.py

```diff
@@ -256,1293 +268,1217 @@ def slots(self) -> List[int]:
+            self.selector,
+        ) = self._prerun_validation(self.mode, self.abi_method_signature)
 
     @classmethod
-    def csv_report(
+    def multi_exec(
         cls,
-        inputs: List[Sequence[PyTypes]],
-        dr_resps: List["DryRunInspector"],
-        txns: Optional[List[Dict[str, Any]]] = None,
-    ) -> str:
-        """Produce a Comma Separated Values report string capturing important statistics
-        for a sequence of dry runs.
+        execs: List["DryRunExecutor"],
+        inputs: Sequence[Sequence[PyTypes]],
+        *,
+        txn_params: Optional[DryRunTransactionParams] = None,
+        verbose: bool = False,
+    ) -> Sequence[Sequence[DryRunInspector]]:
+        return [
+            cast(
+                Sequence[DryRunInspector],
+                e.run(inputs, txn_params=txn_params, verbose=verbose),
+            )
+            for e in execs
+        ]
 
-        For example, assuming you have a string `teal` which is a TEAL program computing $`x^2`$
-        such as this [example program](x/blackbox/teal/app_square.teal).
-        Let's run some Exploratory Dry Run Analysis (EDRA) for $`x`$ in the range $`[0, 10]`$:
-
-        ```python
-        >>> algod = get_algod()
-        >>> inputs = [(x,) for x in range(11)]  # [(0), (1), ... , (10)]
-        >>> run_results = DryRunExecutor.dryrun_app_on_sequence(algod, teal, inputs)
-        >>> csv = DryRunInspector.csv_report(inputs, run_results)
-        >>> print(csv)
-        ```
-        Then you would get the following output:
-        ```plain
-        Run, Status, cost, final_message, last_log, top_of_stack,Arg_00,max_stack_height,s@000,s@001,steps
-        1,REJECT,14,REJECT,`None,0,0,2,,,15
-        2,PASS,14,PASS,`0000000000000001,1,1,2,1,1,15
-        3,PASS,14,PASS,`0000000000000004,4,2,2,2,4,15
-        4,PASS,14,PASS,`0000000000000009,9,3,2,3,9,15
-        5,PASS,14,PASS,`0000000000000010,16,4,2,4,16,15
-        6,PASS,14,PASS,`0000000000000019,25,5,2,5,25,15
-        7,PASS,14,PASS,`0000000000000024,36,6,2,6,36,15
-        8,PASS,14,PASS,`0000000000000031,49,7,2,7,49,15
-        9,PASS,14,PASS,`0000000000000040,64,8,2,8,64,15
-        10,PASS,14,PASS,`0000000000000051,81,9,2,9,81,15
-        ```
+    def run(
+        self,
+        inputs: Union[Sequence[Sequence[PyTypes]], Sequence[PyTypes]],
+        *,
+        txn_params: Optional[DryRunTransactionParams] = None,
+        verbose: bool = False,
+    ) -> Union[Sequence[DryRunInspector], DryRunInspector]:
         """
-        N = len(inputs)
-        assert N == len(
-            dr_resps
-        ), f"cannot produce CSV with unmatching size of inputs ({len(inputs)}) v. drresps ({len(dr_resps)})"
-        if txns:
-            assert N == len(
-                txns
-            ), f"cannot produce CSV with unmatching size of inputs ({len(inputs)}) v. txns ({len(txns)})"
+        TODO: eliminate type-switch on inputs using
+        functools: singledispatch + partial
```

Comment:
@tzaffi Sanity-checking PR scope:  Is the highlighted TODO intended to be addressed in-scope to the PR?

Based on the related _issue_, it seems like _yes_.  Though it might be the case that you're looking to hold off due to the size of changes already involved.

Reply:
This is more of "a wish" than a todo. After writing this I decided internally not to pursue for now. But I forgot to remove the TODO. I'll remove it

Reply:
I've removed it